http/go/nethttp/compression_handler.go

```diff
@@ -0,0 +1,85 @@
+package nethttplibrary
+
+import (
+	"bytes"
+	"compress/gzip"
+	"io"
+	"io/ioutil"
+	"net/http"
+)
+
+type CompressionHandler struct {
+	options CompressionOptions
+}
+
+type CompressionOptions struct {
```

Comment:
missing interface implementation https://github.com/microsoft/kiota/blob/3d240712a5923c628da5c0faf5560284516d8758/http/go/nethttp/redirect_handler.go#L52
https://github.com/microsoft/kiota/blob/main/abstractions/go/request_option.go

Reply:
@baywet why does GetKey return a **RequestOptionKey** instead of **string**?

Reply:
https://pauldigian.com/golang-and-context-an-explanation

> Beware that as a key we should not use a default type (like string). This is to avoid having multiple packages, from different developers, overwrites each other keys.

At the time I read recommendations like there at a couple of different places.